scripts/analysis.py

```diff
@@ -638,50 +583,171 @@ def final_stockpile(cursor, facility):
+    fissile_in_spent = sum(
+        [massfrac for (nucid, massfrac) in out_recipe if nucid in fissile_list])
+    in_fissile = sum([massfrac for (nucid, massfrac)
+                      in in_recipe if nucid in fissile_list])
+    print('The Conversion Factor is:')
+    print((FP + fissile_in_spent - in_fissile) / FP)
+
+
+def mix_ratio(cursor, fuel_recipe_name, spent_fuel_recipe_name, depleted_u_recipe_name, what_reprocess):
+    """ Finds the mixing ratio of separated material and depleted Uranium
 
-        try:
-            quantity_timeseries = get_timeseries(fuel_quantity, duration, .001)
-            fuel_dict[fuel] = quantity_timeseries
-        except:
-            print(str(fuel) + ' has not been used.')
+    Parameters
+    ----------
+    cursor: sqlite cursor
+        sqlite cursor
+    fuel_recipe_name: str
+        name of desired fuel recipe
+    spent_fuel_recipe_name: str
+        name of spent fuel recipe
+    depleted_u_recipe_name: str
+        name of depleted uranium recipe
+    what_reprocess: list
+        list of what elements are separated [zz, zz, zz]
 
+    Returns
+    -------
+    prints ratio of separated material to depleted uranium
+    """
+    query = ('SELECT nucid, massfrac FROM recipes '
+             'INNER JOIN compositions '
+             'ON recipes.qualid = compositions.qualid '
+             'WHERE recipe = "dummy"')
+    fuel_recipe = cursor.execute(query.replace(
+        'dummy', fuel_recipe_name)).fetchall()
+    spent_fuel_recipe = cursor.execute(query.replace(
+        'dummy', spent_fuel_recipe_name)).fetchall()
+    depleted_u_recipe = cursor.execute(query.replace(
+        'dummy', depleted_u_recipe_name)).fetchall()
+    sep_matl = [[nucid, massfrac] for (
+        nucid, massfrac) in spent_fuel_recipe if int(nucid / 10000000) in what_reprocess]
+
+    ratio_list = np.arange(0, 1, .001)
+    prev_err = 1
+    optimal_ratio = 0
+    for ratio in ratio_list:
+        total_err = 0
+        for t in fuel_recipe:
+            reprocessed = sum([massfrac for (nucid, massfrac)
+                               in sep_matl if nucid == t[0]]) * ratio
+            uranium = sum([massfrac for (nucid, massfrac)
+                           in depleted_u_recipe if nucid == t[0]]) * (1 - ratio)
+            value = reprocessed + uranium
+            err = abs(value - t[1])
+            total_err += err
+        if prev_err > total_err:
+            optimal_ratio = ratio
+            prev_err = total_err
+    print('The Optimal Ratio is:')
+    print(optimal_ratio)
+    for t in fuel_recipe:
+        reprocessed = sum([massfrac for (nucid, massfrac)
+                           in sep_matl if nucid == t[0]]) * optimal_ratio
+        uranium = sum([massfrac for (nucid, massfrac)
+                       in depleted_u_recipe 
+                       if nucid == t[0]]) * (1 - optimal_ratio)
+        value = reprocessed + uranium
+        err = abs(value - t[1])
+        print('Error for ' + str(t[0]) + ': ' +
```

Comment:
If I am not mistaken, I think you are referring to a different type of error.

Error here is the composition difference between the mixed fuel and the desired fuel recipe.
The script shows how much off the user will be (per isotope) if he/she uses the mixer ratio output by the function.

If I understood thing wrongly, please let me know!